src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -241,7 +262,10 @@ private void constructSearchQueryRecord(final SearchPhaseContext context, final
                 );
             }
 
-            String hashcode = QueryShapeGenerator.getShapeHashCodeAsString(request.source(), false);
+            String hashcode = "";
```

Comment:
Let's just don't add `QUERY_HASHCODE` as an attribute if it is not enabled. There's no point to add an empty string to the attribute map.